ansible_rulebook/terminal.py

```diff
@@ -0,0 +1,191 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import datetime
+import logging
+import os
+import pprint
+import re
+import threading
+import typing
+
+logger = logging.getLogger(__name__)
+
+
+###############################################################################
+class _Singleton(type):
```

Comment:
If we are going to provide a singleton metaclass would not be better to put it in a different place?

Reply:
With only a single consumer of it there's no specific need.  When/if there is more than one consumer that would make sense.